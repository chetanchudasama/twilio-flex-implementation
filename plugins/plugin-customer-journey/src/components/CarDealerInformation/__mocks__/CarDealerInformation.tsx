import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
