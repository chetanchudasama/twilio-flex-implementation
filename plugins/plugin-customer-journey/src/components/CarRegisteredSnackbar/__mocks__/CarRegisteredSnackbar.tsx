import React from "react";
import { CarRegisteredSnackbarProps } from "../CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = (
  props
) => {
  return <div className="car-registered-snackbar">CarRegisteredSnackbar</div>;
};
