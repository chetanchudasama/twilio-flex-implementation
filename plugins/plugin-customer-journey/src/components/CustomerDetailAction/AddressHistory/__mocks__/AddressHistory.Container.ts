import AddressHistory from "./AddressHistory";

export default AddressHistory;
