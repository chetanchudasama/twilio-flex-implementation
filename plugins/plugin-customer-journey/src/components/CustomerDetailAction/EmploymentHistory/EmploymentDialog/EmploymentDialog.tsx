import React, { useState, useMemo } from "react";
import { v4 } from "uuid";

import {
  CustomerEmploymentModel,
  CustomInput,
  CustomSelect,
  ResponsiveDialog,
  Shared,
  BaseAddressModel,
  OccupationStatusModel,
} from "@common/components";
import {
  Checkbox,
  FormControlLabel,
  Grid,
  MenuItem,
  Typography,
  InputLabel,
} from "@material-ui/core";

import { EmploymentDialogProps } from "./EmploymentDialog.Props";
import {
  EmploymentDialogStyles,
  ResponsiveDialogContentStyles,
} from "./EmploymentDialog.Styles";
import Loading from "../../../LoadingComponent/Loading";
import { AddressSearch } from "../../../AddressSearch/AddressSearch";

export const EmploymentDialog: React.FC<EmploymentDialogProps> = ({
  open,
  isEdit,
  loading,
  activeEmployment,
  onSave,
  onClose,
  onDelete,
}) => {
  const [errors, setErrors] = useState<Record<string, string[]>>({});
  const [employment, setEmployment] =
    useState<CustomerEmploymentModel>(activeEmployment);
  const [openConfirmation, setOpenConfirmation] = useState<boolean>(false);

  const isEmployed: boolean = useMemo(() => {
    const index = Shared.OccupationStatuses.findIndex(
      (s: OccupationStatusModel) =>
        s.name.toLocaleLowerCase() === "not employed"
    );
    return employment?.occupationStatus?.id !== index + 1;
  }, [employment?.occupationStatus?.id]);

  const isEmployerAddressExist: boolean = useMemo(() => {
    return !!(
      employment?.employerAddress.postcode ||
      employment?.employerAddress.buildingName ||
      employment?.employerAddress.subBuilding ||
      employment?.employerAddress.buildingNumber ||
      employment?.employerAddress.streetName ||
      employment?.employerAddress.town
    );
  }, [employment?.employerAddress]);

  const onConfirm = () => {
    const errorObj: Record<string, string[]> = {};
    if (employment?.occupationStatus?.id <= 0) {
      errorObj.occupationStatus = ["Occupation status is required"];
    }

    if (
      employment.yearsAtEmployment === 0 &&
      employment.monthsAtEmployment === 0
    ) {
      errorObj.employmentTime = ["Experience is required"];
    }
    if (isEmployed) {
      if (employment.occupation === "") {
        errorObj.occupation = ["Occupation is required"];
      }

      if (employment.employerName === "") {
        errorObj.employerName = ["Name of employer is required"];
      }

      if (employment.employerPhoneNumber === "") {
        errorObj.employerPhoneNumber = ["Phone number of employer is required"];
      }

      if (!isEmployerAddressExist) {
        errorObj.employerAddress = ["Address of employer is required"];
      }
    }

    if (!isEmployed && employment.employerAddress.town === "") {
      errorObj.town = ["Town is required"];
    }

    setErrors(errorObj);
    if (Object.keys(errorObj).length > 0) {
      return;
    }

    const employmentDetail = { ...employment };
    // reset fields if occupation status is "Not Employed"
    if (!isEmployed) {
      employmentDetail.occupation = "";
      employmentDetail.employerName = "";
      employmentDetail.employerPhoneNumber = "";
      employmentDetail.employerAddress = new BaseAddressModel();
      employmentDetail.employerAddress.town = employment.employerAddress.town;
    }
    onSave(employmentDetail);
  };

  const onClickDelete = () => {
    setOpenConfirmation(true);
  };

  const onChange =
    (key: keyof CustomerEmploymentModel | keyof BaseAddressModel) =>
    (event: React.ChangeEvent<HTMLSelectElement | HTMLInputElement>) => {
      let updatedValue: any = event.target.value;
      if (key === "occupationStatus") {
        updatedValue = Shared.OccupationStatuses.find(
          (s: OccupationStatusModel) => s.id === Number(updatedValue)
        );
      }
      if (key === "town") {
        setEmployment({
          ...employment,
          employerAddress: {
            ...employment.employerAddress,
            [key]: updatedValue,
          },
        });
        return;
      }
      setEmployment({
        ...employment,
        [key]: updatedValue,
      });
    };

  const onCheck = (
    event: React.ChangeEvent<HTMLSelectElement | HTMLInputElement>,
    checked: boolean
  ) => {
    event.stopPropagation();
    setEmployment({
      ...employment,
      isCurrentEmployment: checked,
    });
  };

  const handleUpdateAddress = (baseAddressModel: BaseAddressModel) => {
    setEmployment({
      ...employment,
      employerAddress: {
        ...employment.employerAddress,
        buildingName: baseAddressModel.buildingName,
        buildingNumber: baseAddressModel.buildingNumber,
        postcode: baseAddressModel.postcode,
        streetName: baseAddressModel.streetName,
        subBuilding: baseAddressModel.subBuilding,
        town: baseAddressModel.town,
      },
    });
  };

  return (
    <EmploymentDialogStyles item xs={12}>
      {open && (
        <ResponsiveDialog
          title={isEdit ? "Edit Employment" : "Add Employment"}
          open={open}
          onCancel={onClose}
          onConfirm={onConfirm}
          maxWidth="sm"
          showActionButtons
          showDeleteButton={isEdit}
          onDelete={onClickDelete}
          okText="Save Employment"
          cancelText="Cancel"
          deleteText="Delete Employment"
        >
          <ResponsiveDialogContentStyles item xs={12}>
            <Grid
              container
              direction="row"
              justify="space-between"
              alignItems="center"
              spacing={8}
            >
              <Grid item xs={12}>
                <Typography component="p" className="employment-label">
                  Employment Details
                </Typography>
              </Grid>
              <Grid item xs={12} className="occupation-status">
                <CustomSelect
                  label="Occupation Status"
                  onChange={onChange("occupationStatus")}
                  value={employment.occupationStatus?.id ?? -1}
                  isRequired
                  errors={errors.occupationStatus}
                >
                  <MenuItem value={-1}>
                    <em>Select a status</em>
                  </MenuItem>
                  {Shared.OccupationStatuses.map(
                    (occupationStatus: OccupationStatusModel) => (
                      <MenuItem key={v4()} value={occupationStatus.id}>
                        {occupationStatus.name}
                      </MenuItem>
                    )
                  )}
                </CustomSelect>
              </Grid>
              <Grid item xs={6} className="employment-time">
                <CustomSelect
                  label="Years"
                  onChange={onChange("yearsAtEmployment")}
                  value={employment.yearsAtEmployment ?? 0}
                  isRequired
                  errors={errors.employmentTime}
                >
                  {Array.from(Array(25).keys()).map((n) => (
                    <MenuItem key={v4()} value={n}>
                      {`${n} Years`}
                    </MenuItem>
                  ))}
                </CustomSelect>
              </Grid>
              <Grid item xs={6}>
                <CustomSelect
                  label="Months"
                  onChange={onChange("monthsAtEmployment")}
                  value={employment.monthsAtEmployment ?? 0}
                  isRequired
                  errors={errors.employmentTime}
                >
                  {Array.from(Array(12).keys()).map((n) => (
                    <MenuItem key={v4()} value={n}>
                      {`${n} Months`}
                    </MenuItem>
                  ))}
                </CustomSelect>
              </Grid>
              {isEmployed && (
                <Grid item xs={12} className="emp-occupation">
                  <CustomInput
                    label="Occupation"
                    value={employment.occupation ?? ""}
                    onChange={onChange("occupation")}
                    errors={errors.occupation}
                    isRequired
                  />
                </Grid>
              )}
              <Grid item xs={12}>
                <FormControlLabel
                  className="form-control-label"
                  control={
                    <Checkbox
                      value="currentEmployment"
                      checked={employment.isCurrentEmployment ?? false}
                      onChange={onCheck}
                    />
                  }
                  label="This is their current employment"
                />
              </Grid>
              {isEmployed && (
                <>
                  <Grid item xs={12}>
                    <Typography component="p" className="employment-label">
                      Employer Details
                    </Typography>
                  </Grid>
                  <Grid item xs={12} className="employer-name">
                    <CustomInput
                      label="Employer name"
                      value={employment.employerName ?? ""}
                      onChange={onChange("employerName")}
                      errors={errors.employerName}
                      isRequired
                    />
                  </Grid>
                  <Grid item xs={12} className="employer-phone-no">
                    <CustomInput
                      label="Employer Phone No"
                      value={employment.employerPhoneNumber ?? ""}
                      onChange={onChange("employerPhoneNumber")}
                      errors={errors.employerPhoneNumber}
                      isRequired
                    />
                  </Grid>
                  <Grid item xs={12} className="employer-address">
                    <InputLabel
                      shrink
                      required
                      error={!!errors.employerAddress}
                    >
                      Employer Address
                    </InputLabel>
                    <AddressSearch
                      updateAddress={handleUpdateAddress}
                      address={
                        isEmployerAddressExist
                          ? employment.employerAddress
                          : undefined
                      }
                    />
                    {!!errors.employerAddress && (
                      <p className="employer-address-error">
                        {errors.employerAddress}
                      </p>
                    )}
                  </Grid>
                </>
              )}
              {!isEmployed && (
                <Grid item xs={12} className="town">
                  <CustomInput
                    label="Town"
                    value={employment.employerAddress.town ?? ""}
                    onChange={onChange("town")}
                    errors={errors.town}
                    isRequired
                  />
                </Grid>
              )}
            </Grid>
            {loading && <Loading />}
          </ResponsiveDialogContentStyles>
        </ResponsiveDialog>
      )}
      {openConfirmation && (
        <ResponsiveDialog
          open={openConfirmation}
          maxWidth="xs"
          onConfirm={onDelete}
          onCancel={() => setOpenConfirmation(false)}
          showActionButtons
          okText="Yes"
          cancelText="No"
          isConfirmationDialog
        >
          <div className="confirmation-message">
            Are you sure you want to delete this employment?
          </div>
        </ResponsiveDialog>
      )}
    </EmploymentDialogStyles>
  );
};
