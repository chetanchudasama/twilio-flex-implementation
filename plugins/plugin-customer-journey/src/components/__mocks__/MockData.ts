import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
