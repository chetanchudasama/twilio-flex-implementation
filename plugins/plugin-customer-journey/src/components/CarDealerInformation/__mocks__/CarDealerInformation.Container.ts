import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
