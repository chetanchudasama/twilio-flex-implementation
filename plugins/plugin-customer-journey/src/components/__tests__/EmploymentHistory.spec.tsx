import React from "react";
import { Manager } from "@twilio/flex-ui";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import { create } from "react-test-renderer";
import { CustomerEmploymentModel } from "@common/components";
import { EmploymentListItem } from "../CustomerDetailAction/EmploymentHistory/EmploymentListItem/EmploymentListItem";
import { Props } from "../CustomerDetailAction/EmploymentHistory/EmploymentHistory.Props";
import { EmploymentHistory } from "../CustomerDetailAction/EmploymentHistory/EmploymentHistory";
import { flexManagerMock } from "../__mocks__/MockData";
import { EmploymentDialog } from "../CustomerDetailAction/EmploymentHistory/EmploymentDialog/EmploymentDialog";
import { showErrorMessage, showMessage } from "../../Notifications";

Manager.getInstance = flexManagerMock;

const mockUpdateBaseApplicationDetail = jest.fn();

jest.mock("../../services/application.service", () => ({
  useApplicationService: jest.fn(() => ({
    updateBaseApplicationDetail: mockUpdateBaseApplicationDetail,
  })),
}));

jest.mock("../../Notifications", () => {
  return {
    showErrorMessage: jest.fn(() => {}),
    showMessage: jest.fn(() => {}),
    CustomNotificationType: { SuccessNotification: "successNotification" },
  };
});

const employments: CustomerEmploymentModel[] = [
  {
    occupationStatus: { id: -1, name: "Full-time" },
    yearsAtEmployment: 1,
    monthsAtEmployment: 11,
    occupation: "Car rental agent",
    employerName: "Zara R Woods",
    employerPhoneNumber: "+446349900077",
    employerAddress: {
      postcode: "IP6 3YZ",
      buildingName: "Granary",
      subBuilding: "Granary-23",
      buildingNumber: "2",
      streetName: "75 Golf Road",
      town: "SWILLAND",
    },
    isCurrentEmployment: false,
  },
  {
    occupationStatus: { id: -1, name: "Part-time" },
    yearsAtEmployment: 2,
    monthsAtEmployment: 25,
    occupation: "Car detailer",
    employerName: "Elliot S Naylor",
    employerPhoneNumber: "+446349988077",
    employerAddress: {
      postcode: "IP23 8UJ",
      buildingName: "Haven",
      subBuilding: "Haven-234",
      buildingNumber: "3",
      streetName: "21 Farburn Terrace",
      town: "LITTLE GREEN",
    },
    isCurrentEmployment: true,
  },
];

const props: Props = {
  applicationId: 0,
  currentEmployment: employments[1],
  previousEmployments: [employments[0]],
  setCustomerDetails: jest.fn(),
};

describe("EmploymentHistory", () => {
  it("renders correctly", () => {
    const wrapper = create(<EmploymentHistory {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("renders current employment and previous employment correctly in UI", () => {
    const wrapper = create(<EmploymentHistory {...props} />).root;

    expect(
      wrapper.findAllByProps({
        className: "employment-label",
      })[1].props.children
    ).toContain("Current Employment");
    expect(
      wrapper.findAllByProps({
        className: "employment-label",
      })[2].props.children
    ).toContain("Previous Employment");
  });

  it("renders employment history item correctly in UI", () => {
    const wrapper = mount(<EmploymentHistory {...props} />);

    expect(wrapper.find(EmploymentListItem)).toHaveLength(
      [props.currentEmployment, ...props.previousEmployments].length
    );
  });

  it("should open employment dialog, when add employment button is clicked", async () => {
    const wrapper = mount(<EmploymentHistory {...props} />);

    wrapper.find(".add-employment-btn").at(0).simulate("click");
    wrapper.update();

    expect(wrapper.find(EmploymentDialog).exists()).toBeTruthy();
  });

  it("should open employment dialog, when edit employment button is clicked", async () => {
    const wrapper = mount(<EmploymentHistory {...props} />);

    act(() => {
      wrapper.find(EmploymentListItem).at(0).props().setActiveEmployment();
    });
    wrapper.update();

    expect(wrapper.find(EmploymentDialog).exists()).toBeTruthy();
  });

  it("should close employment dialog, when close button is clicked from EmploymentDialog", async () => {
    const wrapper = mount(<EmploymentHistory {...props} />);

    wrapper.find(".add-employment-btn").at(0).simulate("click");
    wrapper.update();

    expect(wrapper.find(EmploymentDialog).exists()).toBeTruthy();

    act(() => {
      wrapper.find(EmploymentDialog).props().onClose();
    });
    wrapper.update();

    expect(wrapper.find(EmploymentDialog).exists()).toBeFalsy();
  });

  it("should delete employment, when delete button is clicked from EmploymentDialog", async () => {
    mockUpdateBaseApplicationDetail.mockImplementationOnce(() =>
      Promise.resolve(undefined)
    );

    const wrapper = mount(<EmploymentHistory {...props} />);

    act(() => {
      wrapper.find(EmploymentListItem).at(0).props().setActiveEmployment();
    });
    wrapper.update();
    expect(wrapper.find(EmploymentDialog).exists()).toBeTruthy();

    wrapper.find(EmploymentDialog).props().onDelete();
    wrapper.update();

    await act(async () => {
      expect(mockUpdateBaseApplicationDetail).toHaveBeenCalled();
      wrapper.update();
    });

    expect(showMessage).toHaveBeenCalledWith(
      "successNotification",
      "Employment detail deleted successfully!"
    );
  });

  it("should display error message, if delete employment detail API returns an error", async () => {
    mockUpdateBaseApplicationDetail.mockImplementationOnce(() =>
      Promise.reject()
    );

    const wrapper = mount(<EmploymentHistory {...props} />);

    act(() => {
      wrapper.find(EmploymentListItem).at(0).props().setActiveEmployment();
    });
    wrapper.update();

    expect(wrapper.find(EmploymentDialog).exists()).toBeTruthy();

    wrapper.find(EmploymentDialog).props().onDelete();
    wrapper.update();

    await act(async () => {
      expect(mockUpdateBaseApplicationDetail).toHaveBeenCalled();
      wrapper.update();
    });

    expect(showErrorMessage).toHaveBeenCalledWith(
      "Something went wrong, please try again!",
      "",
      true
    );
  });

  it("should update employment, when delete button is clicked from EmploymentDialog", async () => {
    mockUpdateBaseApplicationDetail.mockImplementationOnce(() =>
      Promise.resolve(undefined)
    );

    const wrapper = mount(<EmploymentHistory {...props} />);

    act(() => {
      wrapper.find(EmploymentListItem).at(0).props().setActiveEmployment();
    });
    wrapper.update();

    expect(wrapper.find(EmploymentDialog).exists()).toBeTruthy();

    await act(async () => {
      wrapper.find(EmploymentDialog).props().onSave(employments[1]);
      wrapper.update();

      expect(mockUpdateBaseApplicationDetail).toHaveBeenCalled();
    });

    expect(showMessage).toHaveBeenCalledWith(
      "successNotification",
      "Employment history detail updated successfully!"
    );
  });

  it("should show error message, if update employment detail API returns an error", async () => {
    mockUpdateBaseApplicationDetail.mockImplementationOnce(() =>
      Promise.reject()
    );

    const wrapper = mount(<EmploymentHistory {...props} />);

    act(() => {
      wrapper.find(EmploymentListItem).at(0).props().setActiveEmployment();
    });
    wrapper.update();

    expect(wrapper.find(EmploymentDialog).exists()).toBeTruthy();

    await act(async () => {
      wrapper.find(EmploymentDialog).props().onSave(employments[1]);
      wrapper.update();

      expect(mockUpdateBaseApplicationDetail).toHaveBeenCalled();
    });

    expect(showErrorMessage).toHaveBeenCalledWith(
      "Something went wrong, please try again!",
      "",
      true
    );
  });
});
