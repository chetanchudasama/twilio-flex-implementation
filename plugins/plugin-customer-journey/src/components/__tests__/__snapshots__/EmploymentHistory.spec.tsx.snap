// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmploymentHistory renders correctly 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-trzcga"
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 title"
    >
      Employment History
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 paper-container"
    >
      <div
        className="MuiPaper-root-98 MuiPaper-elevation1-101 MuiPaper-rounded-99 paper"
      >
        <div
          className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
          >
            <p
              className="MuiTypography-root-125 MuiTypography-body1-134 employment-label"
            >
              Current Employment
            </p>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-1es052e"
          >
            <div
              className="MuiCardHeader-root-161 card-header"
            >
              <div
                className="MuiCardHeader-avatar-162"
              >
                <div
                  className="MuiAvatar-root-167 MuiAvatar-colorDefault-168 icon"
                  color="secondary"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-170 MuiSvgIcon-colorSecondary-172"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0zm10 5h4v2h-4zm0 0h4v2h-4z"
                      fill="none"
                    />
                    <path
                      d="M10 16v-1H3.01L3 19c0 1.11.89 2 2 2h14c1.11 0 2-.89 2-2v-4h-7v1h-4zm10-9h-4.01V5l-2-2h-4l-2 2v2H4c-1.1 0-2 .9-2 2v3c0 1.11.89 2 2 2h6v-2h4v2h6c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zm-6 0h-4V5h4v2z"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="MuiCardHeader-content-164"
              >
                <span
                  className="MuiTypography-root-125 MuiTypography-body2-133 MuiCardHeader-title-165"
                >
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 title title-line-1"
                  >
                    Car detailer (Part-time)
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 title title-line-2"
                  >
                    Elliot S Naylor
                  </span>
                </span>
                <span
                  className="MuiTypography-root-125 MuiTypography-body2-133 MuiTypography-colorTextSecondary-158 MuiCardHeader-subheader-166 subheader"
                >
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    3, Haven Haven-234, 21 Farburn Terrace
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    LITTLE GREEN, IP23 8UJ
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    +446349988077
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    2 years, 25 months
                  </span>
                </span>
              </div>
              <div
                className="MuiCardHeader-action-163"
              >
                <button
                  className="MuiButtonBase-root-205 MuiButton-root-179 MuiButton-text-181 MuiButton-textSecondary-183 MuiButton-flat-184 MuiButton-flatSecondary-186 MuiButton-sizeSmall-202 action-btn"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-180"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-170 MuiSvgIcon-fontSizeSmall-177 action-icon"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                    Edit
                  </span>
                  <span
                    className="MuiTouchRipple-root-213"
                  />
                </button>
              </div>
            </div>
            <hr
              className="MuiDivider-root-208 divider"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root-98 MuiPaper-elevation1-101 MuiPaper-rounded-99 paper"
      >
        <div
          className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
        >
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
          >
            <p
              className="MuiTypography-root-125 MuiTypography-body1-134 employment-label"
            >
              Previous Employment
            </p>
          </div>
          <div
            className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 css-1es052e"
          >
            <div
              className="MuiCardHeader-root-161 card-header"
            >
              <div
                className="MuiCardHeader-avatar-162"
              >
                <div
                  className="MuiAvatar-root-167 MuiAvatar-colorDefault-168 icon"
                  color="secondary"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-170 MuiSvgIcon-colorSecondary-172"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0zm10 5h4v2h-4zm0 0h4v2h-4z"
                      fill="none"
                    />
                    <path
                      d="M10 16v-1H3.01L3 19c0 1.11.89 2 2 2h14c1.11 0 2-.89 2-2v-4h-7v1h-4zm10-9h-4.01V5l-2-2h-4l-2 2v2H4c-1.1 0-2 .9-2 2v3c0 1.11.89 2 2 2h6v-2h4v2h6c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zm-6 0h-4V5h4v2z"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="MuiCardHeader-content-164"
              >
                <span
                  className="MuiTypography-root-125 MuiTypography-body2-133 MuiCardHeader-title-165"
                >
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 title title-line-1"
                  >
                    Car rental agent (Full-time)
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 title title-line-2"
                  >
                    Zara R Woods
                  </span>
                </span>
                <span
                  className="MuiTypography-root-125 MuiTypography-body2-133 MuiTypography-colorTextSecondary-158 MuiCardHeader-subheader-166 subheader"
                >
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    2, Granary Granary-23, 75 Golf Road
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    SWILLAND, IP6 3YZ
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    +446349900077
                  </span>
                  <span
                    className="MuiTypography-root-125 MuiTypography-body1-134 subheader-line"
                  >
                    1 years, 11 months
                  </span>
                </span>
              </div>
              <div
                className="MuiCardHeader-action-163"
              >
                <button
                  className="MuiButtonBase-root-205 MuiButton-root-179 MuiButton-text-181 MuiButton-textSecondary-183 MuiButton-flat-184 MuiButton-flatSecondary-186 MuiButton-sizeSmall-202 action-btn"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-180"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-170 MuiSvgIcon-fontSizeSmall-177 action-icon"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                    Edit
                  </span>
                  <span
                    className="MuiTouchRipple-root-213"
                  />
                </button>
              </div>
            </div>
            <hr
              className="MuiDivider-root-208 divider"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 add-employment-container"
    >
      <button
        className="MuiButtonBase-root-205 MuiButton-root-179 MuiButton-contained-190 MuiButton-containedSecondary-192 MuiButton-raised-193 MuiButton-raisedSecondary-195 add-employment-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-180"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-170 MuiSvgIcon-fontSizeSmall-177 action-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          Add Employment
        </span>
        <span
          className="MuiTouchRipple-root-213"
        />
      </button>
    </div>
  </div>
</div>
`;
